memdx/client.go

```diff
@@ -129,13 +126,17 @@ func (c *Client) dispatchCallback(pak *Packet) error {
 		return nil
 	}
 
+	// We delete and then readd the opaque if required.
+	// We do this to prevent a race with Close where both sides can end up
+	// seeing the opaque in the opaque map at the same time.
+	delete(c.opaqueMap, pak.Opaque)
```

Comment:
As discussed on slack, this has issues with racing between handlers and Cancel() calls.  Primarily that if Cancel() is invoked _while_ the handler is running, it won't be in the opaque map and thus does not get cancelled appropriately.